docassemble/assemblylinewizard/data/questions/assembly_line.yml

```diff
@@ -380,24 +381,42 @@ comment: |
   Get the list of fields, and check for any possible labeling errors
   right away
 code: |
+  from pdfminer.pdfparser import PDFSyntaxError
+  from PyPDF2.utils import PdfReadError
+  from zipfile import BadZipFile
+  
   try:
     if template_upload[0].extension == 'pdf':
       all_fields = get_pdf_fields(template_upload)
     else:
       all_fields = get_fields(template_upload)
       
-    if all_fields is None:
-      force_ask('empty_pdf')
+    if not all_fields:
+      empty_pdf
 
     bad_fields = list(filter(
         lambda elem: elem is not None, 
         map(bad_name_reason, all_fields)))
     if len(bad_fields) > 0:
-      force_ask('non_descriptive_field_name')
+      non_descriptive_field_name
       
   except ParsingException as ex:
     parsing_ex = ex
-    force_ask('parsing_exception')
+    parsing_exception
+  
+  except (PDFSyntaxError, PdfReadError, BadZipFile):
+    # Handle PDF read error
+    exit_invalid_file  
+---
+event: exit_invalid_file
+question: |
+  Is this a valid file?
+subquestion: |
+  Docassemble was not able to read the file you uploaded.
+  
+  Make sure it is a valid document.
```

Comment:
Debating if we should show any extra information to the users. They probably won't know what to do with it (I wouldn't either), but if a student reached out to me for help, I'd definitely need the info. Google is always useful too. 

Could also try suggest they run things through qpdf or the documate site again, and add links to those tools here.